ui/components/MesheryPatternServiceForm.js

```diff
@@ -0,0 +1,206 @@
+// @ts-check
+import React from "react";
+import { Tab, Tabs, AppBar, Typography, IconButton } from "@material-ui/core";
+import { Close, Delete } from "@material-ui/icons";
+import PatternService from "./MesheryMeshInterface/PatternService"
+import useStateCB from "../utils/hooks/useStateCB";
+import PascalCaseToKebab from "../utils/PascalCaseToKebab";
+
+function TabPanel(props) {
+  const {
+    children, value, index, ...other
+  } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Typography>{children}</Typography>
+      )}
+    </div>
+  );
+}
+
+function a11yProps(index) {
+  return {
+    id : `simple-tab-${index}`,
+    "aria-controls" : `simple-tabpanel-${index}`,
+  };
+}
+
+function getPatternAttributeName(jsonSchema) {
+  return jsonSchema?._internal?.patternAttributeName || "NA";
+}
+
+function recursiveCleanObject(obj) {
+  for (const k in obj) {
+    if (!obj[k] || typeof obj[k] !== "object") continue;
+
+    recursiveCleanObject(obj[k]);
+
+    if (Object.keys(obj[k]).length === 0) delete obj[k];
+  }
+}
+
+function recursiveCleanObjectExceptEmptyArray(obj) {
+
+  for (const k in obj) {
+    if (!obj[k] || typeof obj[k] !== "object" || Array.isArray(obj[k])) continue;
+
+    recursiveCleanObjectExceptEmptyArray(obj[k]);
+
+    if (Object.keys(obj[k]).length === 0) delete obj[k];
+  }
+}
+
+/**
+ * createPatternFromConfig will take in the form data
+ * and will create a valid pattern from it
+ *
+ * It will/may also perform some sanitization on the
+ * given inputs
+ * @param {*} config
+ */
+function createPatternFromConfig(config, namespace, partialClean = false) {
+  const pattern = {
+    name : `pattern-${Math.random().toString(36).substr(2, 5)}`,
+    services : {},
+  };
+
+  partialClean ? recursiveCleanObjectExceptEmptyArray(config) : recursiveCleanObject(config);
+
+  Object.keys(config).forEach((key) => {
+    // Add it only if the settings are non empty or "true"
+    if (config[key].settings) {
+      const name = PascalCaseToKebab(key);
+      pattern.services[name] = config[key];
+
+      pattern.services[name].type = key;
+      pattern.services[name].namespace = namespace;
+    }
+  });
+
+  Object.keys(pattern.services).forEach((key) => {
+    // Delete the settings attribute/field if it is set to "true"
+    if (pattern.services[key].settings === true) delete pattern.services[key].settings;
```

Comment:
nitpick: remove the === true

Reply:
maybe just to `pattern.services[key].settings` or `!!pattern.services[key].settings`

Reply:
Done.